oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtAuthenticationToken.java

```diff
@@ -36,13 +36,27 @@
 public class JwtAuthenticationToken extends AbstractOAuth2TokenAuthenticationToken<Jwt> {
 	private static final long serialVersionUID = SpringSecurityCoreVersion.SERIAL_VERSION_UID;
 
+	private String name;
+
 	/**
 	 * Constructs a {@code JwtAuthenticationToken} using the provided parameters.
 	 *
 	 * @param jwt the JWT
 	 */
 	public JwtAuthenticationToken(Jwt jwt) {
 		super(jwt);
+		this.name = this.getToken().getSubject();
```

Comment:
This make sense, since `getToken()` is `final`, but what would you think of just doing:

```suggestion
		this.name = jwt.getSubject();
```

So that readers of this code have fewer methods to jump through to understand the code.